lib/iris/fileformats/grib/_save_rules.py

```diff
@@ -200,13 +199,23 @@ def shape_of_the_earth(cube, grib):
 
     # Spherical earth.
     if ellipsoid.inverse_flattening == 0.0:
-        gribapi.grib_set_long(grib, "shapeOfTheEarth", 1)
-        gribapi.grib_set_long(grib, "scaleFactorOfRadiusOfSphericalEarth", 0)
+        # Set shapeOfTheEarth key based on value of spherical earth radius.
+        if int(ellipsoid.semi_major_axis) == 6367470:
+            gribapi.grib_set_long(grib, "shapeOfTheEarth", 0)
+        elif int(ellipsoid.semi_major_axis) == 6371229:
+            gribapi.grib_set_long(grib, "shapeOfTheEarth", 6)
+        else:
+            gribapi.grib_set_long(grib, "shapeOfTheEarth", 1)
+        gribapi.grib_set_long(grib,
+                              "scaleFactorOfRadiusOfSphericalEarth", 0)
         gribapi.grib_set_long(grib, "scaledValueOfRadiusOfSphericalEarth",
                               ellipsoid.semi_major_axis)
     # Oblate spheroid earth.
     else:
-        gribapi.grib_set_long(grib, "shapeOfTheEarth", 7)
+        if isinstance(cs, iris.coord_systems.OSGB):
+            gribapi.grib_set_long(grib, "shapeOfTheEarth", 9)
+        else:
+            gribapi.grib_set_long(grib, "shapeOfTheEarth", 7)
```

Comment:
this appears to me to be due to a default in Cartopy, which expects the units to be metres, not km

However this doesn't appear to be clear in the docstrings
https://github.com/SciTools/iris/pull/1491
I will try to get clarity on this